compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2771,20 +2771,22 @@ class Typer extends Namer
    *  If all this fails, error
    *  Parameters as for `typedUnadapted`.
    */
-  def adapt(tree: Tree, pt: Type, locked: TypeVars)(using Context): Tree =
-    trace(i"adapting $tree to $pt", typr, show = true) {
+  def adapt(tree: Tree, pt: Type, locked: TypeVars, tryGadtHealing: Boolean = true)(using Context): Tree = {
+    val last = Thread.currentThread.getStackTrace()(2).toString;
```

Comment:
@radeusgd one final bit of work: this should 100% be a `def`. Getting a stack trace is costly and `adapt` is, I'd hazard the guess, called very often. Everything else looks good. Thanks again for your excellent work on this PR!

Reply:
Now that I look at it, I think it was just a part of my debugging and possibly can be removed altogether.

Reply:
I have double checked and this stack trace here wasn't very useful (probably that's why I forgot to remove it, as I wasn't using it much during debugging), I have removed it from this trace and made (tryGadtHealing=???) appear only if it's false (because most of the time it's true and it would just clutter the traces), is that ok?